src/type-extensions.ts

```diff
@@ -0,0 +1,13 @@
+import "hardhat/types/config";
+
+import { DlDeployConfig, DlDeployUserConfig } from "./types";
+
+declare module "hardhat/types/config" {
+  interface HardhatUserConfig {
```

Comment:
Let's change the order of these interfaces to math the order in the `src/types.ts` order